File: 0456-132-pattern/0456-132-pattern.java
class Solution {
    public boolean find132pattern(int[] nums) {
        int least = Integer.MIN_VALUE;
        Stack<Integer> stack = new Stack<>();

        for (int i = nums.length - 1; i >= 0; i--) {
            if (nums[i] < least) return true;
            while (!stack.isEmpty() && nums[i] > stack.peek()) {
                least = stack.pop();
            }
            stack.push(nums[i]);
        }

        return false;
    }
}

File: 0456-132-pattern/0456-132-pattern.cs
public class Solution {
    public bool Find132pattern(int[] nums) {
        int least = int.MinValue;
        Stack<int> st = new Stack<int>();

        for (int i = nums.Length - 1; i >= 0; i--) {
            if (nums[i] < least) return true;
            while (st.Count > 0 && nums[i] > st.Peek()) {
                least = st.Pop();
            }
            st.Push(nums[i]);
        }

        return false;
    }
}